src/data.jl

```diff
@@ -119,6 +119,13 @@ SpatialData(domain::𝒟, table::𝒯) where {𝒟,𝒯} =
 domain(sdata::SpatialData) = sdata.domain
 values(sdata::SpatialData) = sdata.table
 
+"""
+    asarray(d::SpatialData{<:RegularGrid}, field)
+
+Return the data of `field` as an `Array`.
+"""
+asarray(d::SpatialData{<:RegularGrid}, field) = reshape(d[field], size(domain(d)))
```

Comment:
I wonder if we should also provide a method for the general domain case. This method would simply return a flat vector without reshaping, makes sense?

Other than that, please use the same identifiers in the other implementations, i.e. `sdata` for spatial data and `var` for spatial variable. The docstring could say something like "Returns the variable `var` as a Julia array with size equal to the size of the underlying domain." and then provide an example with regular grids.

Also, I wonder if this should be part of the public interface for `AbstractData` more generally, and not just the particular instance `SpatialData` implemented in the package.

Reply:
I don't really have an understanding of the datatypes.  So, you'll have to be specific on what types I have to add the method to.

Reply:
We have `AbstractData` to represent any spatial data handled by the project. In particular, we provide two concrete implementations `SpatialData` and `DataView` (should probably be renamed to `GeoData` and `GeoDataView` in the future).

So we could provide a fallback implementation to the abstract parent type:

```julia
asarray(sdata::AbstractData, var) = sdata[var]
```

that returns flat vectors, and then specialize for regular grids as you did with:

```julia
asarray(sdata::SpatialData{<:RegularGrid}, var) = reshape(sdata[var], size(domain(sdata)))
```

The `DataView` case will call the fallback as there is no guarantee that a view preserves regularity of samples. At least that is how I see this `asarray` functionality currently.

Reply:
Regarding: "We have AbstractData to represent any spatial data handled by the project. In particular, we provide two concrete implementations SpatialData and DataView (should probably be renamed to GeoData and GeoDataView in the future)."

I do use GeoData.jl together with GeoStats.jl, so that rename would be a bit inconvenient, just my 2c.

Reply:
I guess the data type in GeoData.jl is actually `GeoArray` right? I need to go over the entire GeoStats.jl stack to make these names more consistent with a `Geo` prefix. AFAIU, Julia can differentiate the module `GeoData` from the data type `GeoStats.GeoData`, that shouldn't be an issue. Please correct me if I am wrong.

Reply:
Consider
```
julia> abstract type GeoStatsBase end

julia> using GeoStatsBase
WARNING: using GeoStatsBase.GeoStatsBase in module Main conflicts with an existing identifier.
ERROR: importing GeoStatsBase into Main conflicts with an existing identifier
```
which would happen analogously if GeoStats.jl was imported before GeoData.jl.  If the other way round, then GeoData.jl would shadow GeoStatsBase.GeoArray.  However, with Julia 1.6 packages can be renamed on import: https://github.com/JuliaLang/julia/blob/master/HISTORY.md

Yes, the datatype is `GeoArray`.

Reply:
The real underlying issue is that GeoData.jl should be named GeoArrays.jl as it defines array types, and not general geospatial data. I will think about this prefix issue, perhaps `Spatial` is enough, but I also dislike the word `Spatial` because it is not the same as `Geospatial`.